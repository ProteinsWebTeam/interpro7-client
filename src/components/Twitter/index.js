import React, { PureComponent } from 'react';

import cancelable from 'utils/cancelable';
import loadResource from 'utils/loadResource';
import Link from 'components/generic/Link';

import getsInView from 'utils/getsInView';

import { foundationPartial } from 'styles/foundation';

import ipro from 'styles/interpro-new.css';
import ebiGlobalStyles from 'ebi-framework/css/ebi-global.scss';
import fonts from 'EBI-Icon-fonts/fonts.css';
import local from './style.css';
const f = foundationPartial(ebiGlobalStyles, fonts, ipro, local);

let bound = false;

const noPadding = { padding: 0 };

class Twitter extends PureComponent /*:: <{}> */ {
  /* ::
    _domNode: ?Element
    _node: ?Element
    _twitterScript: ?{
      cancel: function,
      promise: Promise<any>,
    }
  */
  async componentDidMount() {
    try {
      await getsInView(this._domNode, { rootMargin: '10px' });
      this._twitterScript = cancelable(
        loadResource('//platform.twitter.com/widgets.js'),
      );
      await this._twitterScript.promise;
      if (!window.twttr) return;
      if (!bound) {
        // Only need to bind this once
        window.twttr.events.bind('rendered', ({ target }) => {
          target.style.opacity = 1;
          target.style.transform = 'translateX(0)';
        });
        bound = true;
      }
      if (this._node) window.twttr.widgets.load(this._node);
    } catch (_) {
      /**/
    }
  }

  componentWillUnmount() {
    if (this._twitterScript) this._twitterScript.cancel();
  }

  render() {
    return (
      <div className={f('expanded', 'row')}>
        <div className={f('columns')} style={noPadding}>
          <div className={f('jumbo-news')}>
            <div
              className={f('jumbo-news-container')}
              ref={node => (this._domNode = node)}
            >
              <div
                className={f('icon', 'icon-socialmedia', 'icon-s2')}
                data-icon="T"
              />
              <Link
                ref={node => (this._node = node)}
                data-dnt="true"
                data-chrome={
                  'nofooter noborders noheader noscrollbar transparent'
                }
                data-tweet-limit="1"
                className={f('twitter-timeline')}
                href="https://twitter.com/InterProDB"
                target="_blank"
              >
                Tweets by @InterProDB
              </Link>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Twitter;
