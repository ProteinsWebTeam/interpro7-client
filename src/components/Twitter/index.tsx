import React from 'react';

import Link from 'components/generic/Link';

import cssBinder from 'styles/cssBinder';

import fonts from 'EBI-Icon-fonts/fonts.css';
import local from './style.css';
const css = cssBinder(fonts, local);

const Twitter = () => {
  return (
    <div className={css('twitter-block', 'callout')}>
      <div className={css('icon', 'icon-common')} data-icon="&#xf099;" />
      <Link
        data-dnt="true"
        data-chrome={'nofooter noborders noheader noscrollbar transparent'}
        data-tweet-limit="1"
        className={css('twitter-timeline')}
        href="https://twitter.com/InterProDB"
        target="_blank"
      >
        Tweets by @InterProDB
      </Link>
    </div>
  );
};

export default Twitter;
