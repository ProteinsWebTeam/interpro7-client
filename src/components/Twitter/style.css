@import "../../styles/colors.css";
@import "../../styles/spacing.css";
@import "../../styles/fonts.css";

.test {color:limegreen}

