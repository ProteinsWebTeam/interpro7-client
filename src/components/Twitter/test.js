// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Twitter from '.';

const renderer = new ShallowRenderer();

describe('<Twitter />', () => {
  test('should render', () => {
    renderer.render(<Twitter />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
