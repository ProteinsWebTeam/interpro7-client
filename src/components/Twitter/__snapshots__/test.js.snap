// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Twitter /> should render 1`] = `
<div
  className="expanded row"
>
  <div
    className="columns"
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <div
      className="jumbo-news"
    >
      <div
        className="jumbo-news-container"
      >
        <div
          className="icon icon-common"
          data-icon="ï‚™"
        />
        <Connect(Link)
          className="twitter-timeline"
          data-chrome="nofooter noborders noheader noscrollbar transparent"
          data-dnt="true"
          data-tweet-limit="1"
          href="https://twitter.com/InterProDB"
          target="_blank"
        >
          Tweets by @InterProDB
        </Connect(Link)>
      </div>
    </div>
  </div>
</div>
`;
